resources/js/electron-plugin/src/server/api/childProcess.ts

```diff
@@ -14,16 +16,22 @@ function startProcess(settings) {
         return state.processes[alias];
     }
 
+    const defaultEnv = getDefaultEnvironmentVariables(
+        state.randomSecret,
+        state.electronApiPort
+    );
+
     const proc = utilityProcess.fork(
         join(__dirname, '../../electron-plugin/dist/server/childProcess.js'),
         cmd,
         {
             cwd,
-            serviceName: alias,
             stdio: 'pipe',
+            serviceName: alias,
             env: {
                 ...process.env,
-                ...env,
+                ...defaultEnv,
+                ...env
```

Comment:
I'm keen to understand the order of assignment here to make sure the correct values are applied... I think it's critical that we don't allow the developer to override some of the `defaultEnv` values here

Maybe `defaultEnv` needs to come last?

```suggestion
                ...env,
                ...defaultEnv,
```